### PR TITLE
1544: Implement default_map for indices with more than 3 dimensions

### LINKED ISSUE #1544: Indices over 3-dimensions fail to compile because DefaultMap does not have a backup option
**What Needs to be Done?**

Right now, the SFINAE in `vt::vrt::collection::DefaultMap` fails unless the index is 1-d, 2-d, or 3-d. It should have a backup option.

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #1544

### CODE REVIEW

**Comment by JacobDomagala** [comment]
Codecov issue should be resolved after https://github.com/DARMA-tasking/vt/issues/1736 is fixed.

File: src/vt/topos/mapping/dense/dense.h [hunk 1]
```diff
@@ -74,6 +74,7 @@ using IdxPtr = Index*;
 template <typename T = IdxBase> using Idx1DPtr = IdxType1D<T>*;
 template <typename T = IdxBase> using Idx2DPtr = IdxType2D<T>*;
 template <typename T = IdxBase> using Idx3DPtr = IdxType3D<T>*;
+template <typename T, int8_t N> using IdxNDPtr = vt::index::IdxType<T, N>*;
```
Comment (cz4rs) [inline src/vt/topos/mapping/dense/dense.h:77]: consistency:
```suggestion
template <typename T, int8_t N> using IdxNDPtr = IdxType<T, N>*;
```

**Review by lifflander — APPROVED** [review]
Looks good to me!

File: src/vt/topos/mapping/dense/dense.h (hunk 1 above)
Comment (JacobDomagala) [reply src/vt/topos/mapping/dense/dense.h]: Fixed

**Review by cz4rs — APPROVED** [review]

**Review by jstrzebonski — APPROVED** [review]
Looks good.

**Pull request merged**